zone/gm_commands/parcels.cpp

```diff
@@ -0,0 +1,264 @@
+#include "../client.h"
+#include "../worldserver.h"
+#include "../parcels.h"
+#include "../../common/events/player_events.h"
+
+extern WorldServer worldserver;
+
+void command_parcels(Client *c, const Seperator *sep)
+{
+	const auto arguments = sep->argnum;
+	if (!arguments) {
+		SendParcelsSubCommands(c);
+		return;
+	}
+
+	auto t = c;
+	if (c->GetTarget() && c->GetTarget()->IsClient()) {
+		t = c->GetTarget()->CastToClient();
+	}
+
+	bool is_listdb     = !strcasecmp(sep->arg[1], "listdb");
+	bool is_listmemory = !strcasecmp(sep->arg[1], "listmemory");
+	bool is_details    = !strcasecmp(sep->arg[1], "details");
+	bool is_add        = !strcasecmp(sep->arg[1], "add");
+
+	if (!is_listdb && !is_listmemory && !is_details && !is_add) {
+		SendParcelsSubCommands(c);
+		return;
+	}
+
+	if (is_listdb) {
+		auto player_name = std::string(sep->arg[2]);
+		auto player      = entity_list.GetClientByName(player_name.c_str());
+
+		if (arguments < 2) {
+			c->Message(Chat::White, "Usage: #parcels listdb [Character Name]");
+		}
+		else {
+			if (player_name.empty()) {
+				c->Message(Chat::White, fmt::format("You must provide a player name.").c_str());
+				return;
+			}
+			else {
+				auto results = ParcelsRepository::GetWhere(
+					database, fmt::format("to_name = '{}' ORDER BY slot_id ASC", player_name.c_str()));
+				if (results.empty()) {
+					c->Message(Chat::White, fmt::format("No parcels could be found for {}", player_name).c_str());
```

Comment:
We can return early in here and remove the else below

Reply:
Resolved